synapse/rest/client/v2_alpha/account.py

```diff
@@ -485,12 +519,131 @@ def on_POST(self, request):
         if existing_user_id is not None:
             raise SynapseError(400, "MSISDN is already in use", Codes.THREEPID_IN_USE)
 
+        if not self.hs.config.account_threepid_delegate_msisdn:
+            logger.warn(
+                "No upstream msisdn account_threepid_delegate configured on the server to "
+                "handle this request"
+            )
+            raise SynapseError(
+                400,
+                "Adding phone numbers to user account is not supported by this homeserver",
+            )
+
         ret = yield self.identity_handler.requestMsisdnToken(
-            id_server, country, phone_number, client_secret, send_attempt, next_link
+            self.hs.config.email_account_threepid_delegate_msisdn,
+            country,
+            phone_number,
+            client_secret,
+            send_attempt,
+            next_link,
         )
+
         return 200, ret
 
 
+class AddThreepidSubmitTokenServlet(RestServlet):
+    """Handles 3PID validation token submission for adding a 3PID to a user's account"""
+
+    PATTERNS = client_patterns(
+        "/add_threepid/(?P<medium>[^/]*)/submit_token$", releases=(), unstable=True
```

Comment:
```suggestion
        "/add_threepid/(?P<medium>[^/]*)/submit_token$", releases=(), unstable=True
```

```suggestion
        "/password_reset/(?P<medium>[^/]*)/submit_token$", releases=()
```

Since we're using unstable flags, we can use `r0` for the endpoints.